docs/rfcs/2024-08-06-json-datatype.md

```diff
@@ -0,0 +1,141 @@
+---
+Feature Name: Json Datatype
+Tracking Issue: https://github.com/GreptimeTeam/greptimedb/issues/4230
+Date: 2024-8-6
+Author: "Yuhan Wang <profsyb@gmail.com>"
+---
+
+# Summary
+This RFC proposes a method for storing and querying JSON data in the database.
+
+# Motivation
+JSON is widely used across various scenarios. Direct support for writing and querying JSON can significantly enhance the database's flexibility.
```

Comment:
From a real scenario, ingesting JSON data can invoke a PIPELINE to preprocess JSON data into table structure data which should have better performance and usability. And given we already have the auto-alter mechanism, the flexibility also looks good

Reply:
The JSON can provide an option for users who don't want to write PIPELINE